CHANGELOG.md

```diff
@@ -277,6 +277,9 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/).
 - Pass the `stage` argument of `Callback.{setup,teardown}` as a keyword ([#7973](https://github.com/PyTorchLightning/pytorch-lightning/pull/7973))
 
 
+- Fixed move best score to device in EarlyStopping Callback ([#7959](https://github.com/PyTorchLightning/pytorch-lightning/pull/7959))
```

Comment:
```suggestion
- Fixed move best score to device in EarlyStopping Callback ([#7959](https://github.com/PyTorchLightning/pytorch-lightning/pull/7959))
```
as your comments suggest, this fix applies only for tpu right?

Reply:
It applies to other accelerators as well.

Reply:
how would I reproduce an error on a GPU for example?

why can't we do for example this:

`self.best_score.to(current.device)`

moving one tensor to the device but the other not in a monitor_op is only going to raise questions for the reader of this code

Reply:
```python
>>> torch_inf = torch.tensor(np.Inf)
>>> value = torch.tensor(5, device=xm.xla_device())
>>> torch.lt(value, torch_inf)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
RuntimeError: torch_xla/csrc/aten_xla_bridge.cpp:69 : Check failed: xtensor
*** Begin stack trace ***
	tensorflow::CurrentStackTrace()
```
Similar code for Cuda devices won't throw an error.

Reply:
I got the following error (device with 2 gpus and using ddp)
`RuntimeError:` Expected all tensors to be on the same device, but found at least two devices, cuda:0 and cpu!`

I also felt that `self.best_score.to(current.device)` was safer..